### PR TITLE
[BEAM-4300] Fix ValidatesRunner tests in Python: run with same mechanism as ITs

### PR DESCRIPTION
**Pull request by pabloem**
Changes in this PR:
1. "Python PostCommit" and "Python Dataflow ValidatesRunner" tests now use `run_postcommits.sh` to run.
2. This makes `run_validatesrunner.sh` obsolete, so I remove it.
3. Jenkins job to run ValidatesRunner as part of Gradle
  3.1 Renaming the ValidatesRunner tests to `"beam_PostCommit_Py_VR_Dataflow"`, to have a shorter job name.
4. Small change to `dependency.py` to avoid having `pip` prompt for use input during isntallation of dependencies.

Passing tests:
https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/6/ (ValidatesRunner)
https://builds.apache.org/job/beam_PostCommit_Python_Verify/4997/ (PostCommit)

Seed job before testing: https://builds.apache.org/job/beam_SeedJob/1720/

r: @markflyhigh 
cc: @tvalentyn @udim

### CODE REVIEW

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Python Dataflow ValidatesRunner

**Comment by pabloem** [comment]
Run Python Dataflow ValidatesRunner

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Python Dataflow ValidatesRunner

**Comment by pabloem** [comment]
Running test: https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/2/console

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Python Dataflow ValidatesRunner

**Comment by pabloem** [comment]
Run Python Dataflow ValidatesRunner

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Run Python Dataflow ValidatesRunner

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python Dataflow ValidatesRunner

**Comment by pabloem** [comment]
Run Seed Job

File: sdks/python/run_postcommit.sh [hunk 1]
```diff
@@ -40,11 +39,15 @@ python setup.py sdist
...
+# Install test dependencies for ValidatesRunner tests.
+echo "pyhamcrest" > postcommit_requirements.txt
+echo "mock" >> postcommit_requirements.txt
+
 # Run integration tests on the Google Cloud Dataflow service
 # and validate that jobs finish successfully.
 echo ">>> RUNNING TEST DATAFLOW RUNNER it tests"
 python setup.py nosetests \
-  --attr IT \
+  --attr $1 \
```
Comment (udim) [inline sdks/python/run_postcommit.sh]: Could you verify that this command line option is passed (with a usage blurb if it isn't)?

Comment (pabloem) [reply sdks/python/run_postcommit.sh]: Done. Also added the ability to receive GCP project / path parameters to run tests locally.

**Comment by pabloem** [comment]
Run Python PostCommit

File: sdks/python/run_postcommit.sh [hunk 2]
```diff
@@ -22,17 +22,24 @@
...
+fi
 
 set -e
 set -v
 
 # Run tests on the service.
 
 # Where to store integration test outputs.
-GCS_LOCATION=gs://temp-storage-for-end-to-end-tests
+GCS_LOCATION=${2:-gs://temp-storage-for-end-to-end-tests}
```
Comment (udim) [inline sdks/python/run_postcommit.sh]: Awesome!

**Pull request merged**